R/utils.R

```diff
@@ -0,0 +1,74 @@
+#' Wrapper function to help generate a packrat lockfile
+#'
+#' @param ...      the arguments to pass to gen_runtime_description
+#' @param .workdir directory to pass to install_from_desc
+#' @export
+gen_packrat_lockfile <- function(..., .workdir = file.path( getwd(),"work")){
+  
+  runtime_desc <- pkglock::gen_runtime_description(...)
+  install <- pkglock::install_from_desc(runtime_desc, .dir = .workdir)
+  lockfile <- install$snapshot$lockfile
+  
+  return (lockfile)
+}
+
+#' Scan a library folder and build a package list
+#' @param .libdir the library directory to scan
+#' @export
+build_package_list_from_library_dir <- function(.libdir){
+  
+  pkgs <- list();
+  for( file in list.files(.libdir)){

```

Comment:
whats the chance of a non-pkg file/folder getting into this list? Might want to at least flag this as a TODO for a try().

Rather than checking the names directly, it might just be better to let R handle it by querying the packages directly in `installed.packages(lib.loc = ...)`